addons/binding/org.openhab.binding.nadreceiver/src/main/java/org/openhab/binding/nadreceiver/internal/NadReceiverHandler.java

```diff
@@ -0,0 +1,367 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.nadreceiver.internal;
+
+import static org.openhab.binding.nadreceiver.internal.NadReceiverBindingConstants.*;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ScheduledFuture;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.nadreceiver.internal.protocol.NadReceiverConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NadReceiverHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marc Chételat - Initial contribution
+ */
+public class NadReceiverHandler extends BaseThingHandler implements NadReceiverEventListener {
```

Comment:
Please annotate your classes (except config and protocol) with `@NonNullByDefault` to help prevent NPEs